### PR TITLE
Daily cron job for triggering seqqc runs across Arcadia

### LINKED ISSUE #28: draft of data upload instructions for cron job
New sequencing data should be uploaded to the Arcadia Science S3 bucket `arcadia-seqqc`. Each data set should be uploaded into a new folder in the `indir`. Create a new folder in `arcadia-seqqc/indir`. The folder should be named `<year>-<initials>-<descriptor>` where `descriptor` is an up to 10 character descriptor of your sequencing data. For example, if I created a folder, the full path would be `S3://arcadia-seqqc/indir/2023-ter-timecheese`. New sequencing data needs to be in gzipped FASTQ format (`*fq.gz, *fastq.gz`) and can be single or paired end format. You also need to create and upload a CSV samplesheet that documents your sample names and the paths to the data in the S3 bucket. See [these instructions](https://github.com/Arcadia-Science/seqqc/blob/master/docs/usage.md#samplesheet-input) for how to create and format a sample sheet.

Each (day, week?) the cron job will check whether there is new data in `arcadia-seqqc/indir`. If there is, the cron job will run the seqqc pipeline on the FASTQ files. All FASTQ files in one indir (e.g. `S3://arcadia-seqqc/indir/2023-ter-timecheese`) will be run together at the same time. When the pipeline is finished running, the output files will be available in `S3://arcadia-seqqc/outdir/2023-ter-timecheese`. You will also receive an email with the quality control report attached. This report contains inline documentation for how to interpret the results. If your data are good quality, you can now upload them to the European Nucleotide Archive.

### PR DESCRIPTION
**Pull request by mertcelebi**
Fixes #28

### CODE REVIEW

**Comment by mertcelebi** [comment]
@taylorreiter let me know if we need to cite stuff like, node, Typescript or the AWS SDK.

File: CHANGELOG.md [hunk 1]
```diff
@@ -18,6 +18,7 @@ Initial release of Arcadia-Science/seqqc, created with the [nf-core](https://nf-
 - MultiQC interpretation documentation
 - Test profiles for different sequencing chemistries
 - Updates to email template
+- Added the cron job that automatically triggers the pipeline runs
```
Comment (taylorreiter) [inline CHANGELOG.md]:
```suggestion
- Added the cron job that automatically triggers the pipeline runs using GitHub actions, AWS S3, and AWS Batch via Nextflow Tower
```

File: cron/src/utils.ts [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    // and the other for the actual file name (test.csv). At least, the file name is guaranteed.
+    const csvObjectTokens = csvName.replace(`${AWS_SOURCE_PREFIX}/`, "").split("/");
+
+    // If there are 2 or more tokens, return the first one as the root directory name
+    if (csvObjectTokens.length >= 2) {
+        return csvObjectTokens[0];
+    }
+
+    // If for some reason there is only a single token (ie just the file name)
+    // create a random string for the directory name
```
Comment (taylorreiter) [inline cron/src/utils.ts:40]: clever

**Review by taylorreiter — APPROVED** [review]
This is so exciting, thank you for putting this together! I don't think we need to cite the software you mention above since only arcadians would be interacting with the cron job as a precursor to running the pipeline.

**Pull request merged**